cmd/publicize/server.go

```diff
@@ -0,0 +1,183 @@
+	"k8s.io/test-infra/prow/git/v2"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+)
+
+type githubClient interface {
+	IsMember(org, user string) (bool, error)
+	CreateComment(owner, repo string, number int, comment string) error
+	GetPullRequest(org, repo string, number int) (*github.PullRequest, error)
+}
+
+var publicizeRe = regexp.MustCompile(`(?mi)^/publicize\s*$`)
+
+func helpProvider(_ []prowconfig.OrgRepo) (*pluginhelp.PluginHelp, error) {
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: `The publicize plugin is used for merging and push the commit history to a configured upstream repository.`,
+	}
+	pluginHelp.AddCommand(pluginhelp.Command{
+		Usage:       "/publicize",
+		Description: "Merge the commit histories into the configured upstream repository",
+		WhoCanUse:   "Members of the trusted organization for the repo.",
+		Examples:    []string{"/publicize"},
+	})
+	return pluginHelp, nil
+}
+
+type server struct {
+	githubTokenGenerator func() []byte
+
+	gitName  string
+	gitEmail string
+
+	config func() *Config
+
+	ghc githubClient
+	gc  git.ClientFactory
+
+	secretAgent *secret.Agent
+
+	dry bool
+}
+
+func (s *server) handleIssueComment(l *logrus.Entry, ic github.IssueCommentEvent) {
+	if !publicizeRe.MatchString(ic.Comment.Body) {
+		return
+	}
+
+	org := ic.Repo.Owner.Login
+	repo := ic.Repo.Name
+	num := ic.Issue.Number
+
+	logger := logrus.WithFields(logrus.Fields{
+		github.OrgLogField:  org,
+		github.RepoLogField: repo,
+		github.PrLogField:   num,
+	})
+
+	logger.Info("Publicize of PR has been requested.")
+
+	pr, err := s.ghc.GetPullRequest(org, repo, num)
+	if err != nil {
+		logger.WithError(err).Warn("couldn't get pull request")
+		s.createComment(ic, fmt.Sprintf("couldn't get pull request: %v", err), logger)
+		return
+	}
+	baseBranch := pr.Base.Ref
+
+	if err := s.checkPrerequisites(logger, pr, ic); err != nil {
+		logger.WithError(err).Warn("error occurred while checking for prerequisites")
+		s.createComment(ic, fmt.Sprintf("%v", err), logger)
+		return
+	}
+
+	destOrgRepo := s.config().Repositories[fmt.Sprintf("%s/%s", org, repo)]
+	destOrg := strings.Split(destOrgRepo, "/")[0]
+	destRepo := strings.Split(destOrgRepo, "/")[1]
+	sourceRemoteResolver := func() (string, error) {
+		return fmt.Sprintf("https://%s@%s/%s/%s", string(s.githubTokenGenerator()), github.DefaultHost, org, repo), nil
+	}
+
+	logger.Infof("Trying to merge the PR to destination: %s/%s@%s", destOrg, destRepo, baseBranch)
+	mergeMsg := fmt.Sprintf("merge history from %s/%s", org, repo)
+	if err := s.mergeAndPushToRemote(destOrg, destRepo, sourceRemoteResolver, baseBranch, mergeMsg, s.dry); err != nil {
+		logger.WithError(err).Warnf("couldn't merge the pull request and push to the destination: %v", err)
+		s.createComment(ic, fmt.Sprintf("Publicize failed with error: %v", err), logger)
+		return
+	}
+
+	destOrgRepoLink := fmt.Sprintf("https://%s/%s/tree/%s", github.DefaultHost, destOrgRepo, baseBranch)
```

Comment:
This should also be the host that the user passed in

Reply:
addressed